File: 0101-Symmetric Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    bool isSymmetric(TreeNode* root) {
        return root == nullptr || bfs(root->left, root->right);  // 截断判断
    }
    bool bfs(TreeNode* left, TreeNode* right){
        if(left == nullptr || right == nullptr){
            return left == right;
        }
        if(left->val != right->val){
            return false;
        }
        return bfs(left->left, right->right) && bfs(left->right, right->left);
    }
};

File: 0101-Symmetric Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def isSymmetric(self, root: TreeNode) -> bool:
        return root is None or self.bfs(root.left, root.right)
    
    def bfs(self, left, right):
        if left is None or right is None:
            return left == right
        return False if left.val != right.val else self.bfs(left.left, right.right) and self.bfs(left.right, right.left)